importlib_resources/readers.py

```diff
@@ -135,19 +136,23 @@ class NamespaceReader(abc.TraversableResources):
     def __init__(self, namespace_path):
         if 'NamespacePath' not in str(namespace_path):
             raise ValueError('Invalid path')
-        self.path = MultiplexedPath(*map(self._resolve, namespace_path))
+        self.path = MultiplexedPath(*filter(bool, map(self._resolve, namespace_path)))
 
     @classmethod
-    def _resolve(cls, path_str) -> abc.Traversable:
+    def _resolve(cls, path_str) -> Optional[abc.Traversable]:
         r"""
         Given an item from a namespace path, resolve it to a Traversable.
 
         path_str might be a directory on the filesystem or a path to a
         zipfile plus the path within the zipfile, e.g. ``/foo/bar`` or
         ``/foo/baz.zip/inner_dir`` or ``foo\baz.zip\inner_dir\sub``.
+
+        path_str might also be a sentinel used by editable packages to
+        trigger other behaviors (see python/importlib_resources#311).
+        In that case, return None.
         """
-        (dir,) = (cand for cand in cls._candidate_paths(path_str) if cand.is_dir())
-        return dir
+        dirs = (cand for cand in cls._candidate_paths(path_str) if cand.is_dir())
+        return next(dirs, None)
```

Comment:
Previously, this routine would fail with a `ValueError` if more than one candidate path resolved as a dir. Now it just returns the first one it finds. Is that acceptable, or should we retain the enforcement that no more than one will resolve? I'm leaning toward "it doesn't matter (yet)", so ignore the concern until it manifests, if ever.

Reply:
Is the return value of `cls._candidate_paths` deterministic or may be affected due to how the paths are listed on file?

Reply:
Oh, gosh, I hadn't even looked deeply into what `_candidate_paths` was. When I was writing this note, I was thinking `_candidate_paths` was a potentially long list of paths. I see now it merely generates possible path objects from a path entry. So for `foo/bar`, it will yield `pathlib.Path('foo/bar')` and nothing else. And for `foo/bar.zip/baz`, assuming `foo/bar.zip` is a valid zipfile, it will yield `pathlib.Path('foo/bar.zip/baz')` and `zipfile.Path('foo/bar.zip', 'baz/')`. So it's pretty deterministic... and highly unlikely that `cand.is_dir()` will ever return `True` for more than one thing. I'm now confident this approach is just fine.